### PR TITLE
[PyOV][Tests] Fix some tests for M1

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - *item1*
 - *...*

### Tickets:
 - CVS-98111

### CODE REVIEW

File: src/bindings/python/tests/test_runtime/test_infer_request.py [hunk 1]
```diff
@@ -349,7 +353,7 @@ def check_fill_inputs(request, inputs):
 
 def test_infer_mixed_keys(device):
     core = Core()
-    model = core.read_model(test_net_xml, test_net_bin)
+    model = get_relu_model()
     core.set_property(device, {"PERF_COUNT": "YES"})
     model = core.compile_model(model, device)
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_runtime/test_infer_request.py:358]:
```python
def test_infer_mixed_keys(device):
    compiled_model = get_compiled_model()  # anything that matches two inputs
    
    img = generate_image()
    # ...
```
How about this? Thre is no need to use `PERF_COUNT` and making it device specific.

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 2]
```diff
@@ -30,6 +30,11 @@ def test_property_ro():
 
 def test_allow_auto_batching_property():
     core = Core()
+    device_name = core.get_property("CPU", "FULL_DEVICE_NAME")
+
+    if "Intel" not in device_name:
+        pytest.skip("Properties are Intel(R) CPU specific")
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_runtime/test_properties.py:36]: I think it should be either:
```suggestion
@pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU",
                    reason=f"Cannot run test on device {os.environ.get('TEST_DEVICE')}, Plugin specific test")
def test_allow_auto_batching_property():
    core = Core()
    device_name = core.get_property("CPU", "FULL_DEVICE_NAME")

    if "Intel" not in device_name:
        pytest.skip("Properties are Intel(R) CPU specific")
```
Or:
```suggestion
def test_allow_auto_batching_property(device):
    core = Core()
    device_name = core.get_property(device, "FULL_DEVICE_NAME")

    if device != "CPU" and "Intel" not in device_name:
        pytest.skip("Properties are Intel(R) CPU specific")
```

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 3]
```diff
@@ -78,28 +101,23 @@ def test_single_property_setting():
         "CACHE_DIR": "./",
         properties.inference_num_threads(): 9,
         properties.affinity(): "NONE",
-        "INFERENCE_PRECISION_HINT": Type.f32,
-        properties.hint.performance_mode(): properties.hint.PerformanceMode.LATENCY,
-        properties.hint.num_requests(): 12,
         "NUM_STREAMS": properties.streams.Num(5),
     },
 ])
 def test_properties_core(properties_to_set):
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_runtime/test_properties.py:107]: Can you add just properties tests separetely? Parametrized test for every entry. Let me know if I can take care of it.

File: src/bindings/python/tests/test_utils/test_utils.py [hunk 4]
```diff
@@ -38,14 +29,30 @@ def generate_image(shape: Tuple = (1, 3, 32, 32), dtype: Union[str, np.dtype] =
...
     relu = ops.relu(param, name="relu")
-    model = Model([relu], [param], "test")
+    model = Model([relu], [param], "test_model")
     model.get_ordered_ops()[2].friendly_name = "friendly"
 
+    assert model is not None
+    return model
+
+
+def generate_relu_compiled_model(device, input_shape: List[int] = None) -> openvino.runtime.ie_api.CompiledModel:
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_utils/test_utils.py:44]:
```suggestion
def generate_relu_compiled_model(device, input_shape: List[int] = None) -> openvino.runtime.CompiledModel:
```

Comment (akuporos) [reply src/bindings/python/tests/test_utils/test_utils.py:44]: fixed

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 2 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py:36]: fixed

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 5]
```diff
@@ -43,33 +31,43 @@ def test_get_runtime_model(device):
     assert isinstance(runtime_model, Model)
 
 
-def test_export_import():
+def test_export_import(device):
     core = Core()
+
+    if "EXPORT_IMPORT" not in core.get_property(device, "OPTIMIZATION_CAPABILITIES"):
+        device_name = core.get_property(device, "FULL_DEVICE_NAME")
+        pytest.skip(f"{device_name} plugin due-to export, import model API isn't implemented.")
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:39]: done

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 6]
```diff
@@ -78,9 +76,7 @@ def test_export_import_advanced():
 
 
 def test_get_input_i(device):
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:78]: done

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 7]
```diff
@@ -89,9 +85,7 @@ def test_get_input_i(device):
 
 
 def test_get_input_tensor_name(device):
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:87]: great idea!
done

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 8]
```diff
@@ -111,25 +103,19 @@ def test_get_input(device):
 
 
 def test_get_output_i(device):
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:105]: done

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 9]
```diff
@@ -151,70 +135,54 @@ def test_output_set_friendly_name(device):
...
-    core = Core()
-    model = core.read_model(model=test_net_xml, weights=test_net_bin)
-    compiled_model = core.compile_model(model, device)
+    compiled_model = generate_relu_compiled_model(device)
     outputs = compiled_model.outputs
     assert isinstance(outputs, list)
     assert len(outputs) == 1
 
 
 def test_outputs_items(device):
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:144]: done

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 3 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py:107]: discussed offline: will be fixed by separate PR

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 10]
```diff
@@ -151,70 +135,54 @@ def test_output_set_friendly_name(device):
...
 
 
 def test_input_get_index(device):
-    core = Core()
-    model = core.read_model(model=test_net_xml, weights=test_net_bin)
-    compiled_model = core.compile_model(model, device)
+    compiled_model = generate_relu_compiled_model(device)
     net_input = compiled_model.input(0)
     expected_idx = 0
     assert net_input.get_index() == expected_idx
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:168]: done

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 11]
```diff
@@ -223,9 +191,7 @@ def test_inputs_get_friendly_name(device):
 
 
 def test_inputs_set_friendly_name(device):
-    core = Core()
-    model = core.read_model(model=test_net_xml, weights=test_net_bin)
-    compiled_model = core.compile_model(model, device)
+    compiled_model = generate_relu_compiled_model(device)
     inputs = compiled_model.inputs
     input_0 = inputs[0]
     node = input_0.get_node()
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:197]: done

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 12]
```diff
@@ -235,116 +201,70 @@ def test_inputs_set_friendly_name(device):
...
 def test_inputs_docs(device):
-    core = Core()
-    model = core.read_model(model=test_net_xml, weights=test_net_bin)
-    compiled_model = core.compile_model(model, device)
+    compiled_model = generate_relu_compiled_model(device)
+
     inputs = compiled_model.inputs
     input_0 = inputs[0]
     expected_string = "openvino.runtime.ConstOutput represents port/node output."
     assert input_0.__doc__ == expected_string
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:209]: done

File: src/bindings/python/tests/test_runtime/test_infer_request.py (hunk 1 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_infer_request.py]: removed

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 13]
```diff
@@ -235,116 +201,70 @@ def test_inputs_set_friendly_name(device):
...
     tensor = Tensor(img)
-    compiled_model = core.compile_model(model, device)
     res_tensor = compiled_model.infer_new_request({"data": tensor})
     res_img = compiled_model.infer_new_request({"data": img})
-    assert np.argmax(res_tensor[list(res_tensor)[0]]) == 9
+    assert np.argmax(res_tensor[list(res_tensor)[0]]) == 531
     assert np.argmax(res_tensor[list(res_tensor)[0]]) == np.argmax(res_img[list(res_img)[0]])
 
 
 def test_infer_tensor_numpy_shared_memory(device):
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:190]: added skip, run only on cpu

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 14]
```diff
@@ -7,34 +7,22 @@
...
 
-def test_get_property_model_name(device):
+def test_get_property(device):
+    # todo: for some reason it fails when generate_relu_compiled_model called
+    model = get_relu_model([1, 3, 32, 32])
     core = Core()
-    model = core.read_model(model=test_net_xml, weights=test_net_bin)
-    compiled_model = core.compile_model(model, device)
+    compiled_model = core.compile_model(model, device, {})
     network_name = compiled_model.get_property("NETWORK_NAME")
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py]: discussed, deleted

File: src/bindings/python/tests/test_runtime/test_compiled_model.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:190]: why is it CPU specific? from the first glance the code looks generic

File: src/bindings/python/tests/test_runtime/test_compiled_model.py [hunk 15]
```diff
@@ -151,200 +108,127 @@ def test_output_set_friendly_name(device):
...
-    model = core.read_model(model=xml, weights=weights)
-    img = generate_image()
-    tensor = Tensor(img)
-    compiled_model = core.compile_model(model, device)
-    res = compiled_model.infer_new_request({"data": tensor})
-    assert np.argmax(res[list(res)[0]]) == 9
-
-
+@pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU",
+                    reason=f"Cannot run test on device {os.environ.get('TEST_DEVICE')}, Plugin specific test")
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_compiled_model.py:221]: the same

File: src/bindings/python/tests/test_runtime/test_core.py [hunk 16]
```diff
@@ -283,16 +262,20 @@ def test_query_model(device):
...
+    compiled_model = core.compile_model(model, "BLA")
+    assert isinstance(compiled_model, CompiledModel), "Cannot load the network to the registered plugin with name 'BLA'"
 
 
 @pytest.mark.dynamic_library()
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason="Device independent test")
 def test_register_plugins():
     core = Core()
+    if "Intel" not in core.get_property("CPU", "FULL_DEVICE_NAME"):
+        pytest.skip("This test runs only on openvino intel cpu plugin")
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_core.py:278]: can we have 2 different files for macOS which we will select depending on target platform? For arm64, we can use file with arm plugin, for x86 - current file.

File: src/bindings/python/tests/test_runtime/test_core.py [hunk 17]
```diff
@@ -283,16 +262,20 @@ def test_query_model(device):
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason="Device independent test")
 def test_register_plugin():
     core = Core()
+    if "Intel" not in core.get_property("CPU", "FULL_DEVICE_NAME"):
+        pytest.skip("This test runs only on openvino intel cpu plugin")
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_core.py:266]: let's fix the test with adding if / else depending on platform architecture

File: src/bindings/python/tests/test_runtime/test_infer_request.py [hunk 18]
```diff
@@ -347,21 +351,22 @@ def check_fill_inputs(request, inputs):
     check_fill_inputs(request, inputs)
 
 
+@pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU",
+                    reason=f"Cannot run test on device {os.environ.get('TEST_DEVICE')}, Plugin specific test")
 def test_infer_mixed_keys(device):
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_infer_request.py:356]: may be we need to add info why tests is device specific? Code itself looks generic

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 19]
```diff
@@ -28,8 +28,13 @@ def test_property_ro():
     assert "available_devices(): incompatible function arguments." in str(e.value)
 
 
-def test_allow_auto_batching_property():
+def test_allow_auto_batching_property(device):
     core = Core()
+    device_name = core.get_property(device, "FULL_DEVICE_NAME")
+
+    if device != "CPU" and "Intel" not in device_name:
+        pytest.skip("Properties are Intel(R) CPU specific")
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_properties.py:36]: auto-batching is not CPU specific, it's generic functionality and we should not have any conditional logic here

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 20]
```diff
@@ -41,12 +46,35 @@ def test_allow_auto_batching_property():
                     reason=f"Cannot run test on device {os.environ.get('TEST_DEVICE')}, Plugin specific test")
 def test_single_property_setting():
     core = Core()
+    device_name = core.get_property("CPU", "FULL_DEVICE_NAME")
+    if "Intel" not in device_name:
+        pytest.skip("Properties are Intel(R) CPU specific")
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_properties.py:51]: I suppose this property also works for ARM CPU, GPU, MYRIAD, HDDL

File: src/bindings/python/tests_compatibility/test_utils/test_utils.py [hunk 21]
```diff
@@ -25,13 +25,17 @@ def generate_image(shape: Tuple = (1, 3, 32, 32), dtype: Union[str, np.dtype] =
...
-    core = IECore()
     caps = Function.to_capsule(func)
     cnnNetwork = IENetwork(caps)
+    return cnnNetwork
+
+
+def generate_relu_compiled_model(input_shape: List[int]) -> openvino.inference_engine.ExecutableNetwork:
+    core = IECore()
+    cnnNetwork = generate_relu_model(input_shape)
     return core.load_network(cnnNetwork, "CPU", {})
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests_compatibility/test_utils/test_utils.py:41]: should we replace CPU here with generic device name?

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]

File: src/bindings/python/tests_compatibility/test_utils/test_utils.py (hunk 21 above)
Comment (akuporos) [reply src/bindings/python/tests_compatibility/test_utils/test_utils.py]: done

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 19 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py]: This test file was updated separately here: https://github.com/openvinotoolkit/openvino/pull/14684

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 20 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py]: This test file was updated separately here: https://github.com/openvinotoolkit/openvino/pull/14684

File: src/bindings/python/tests/test_runtime/test_compiled_model.py (hunk 15 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py]: double checked with GPU plugin, this skip may be removed.

File: src/bindings/python/tests/test_runtime/test_compiled_model.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_compiled_model.py:184]: double checked with GPU plugin, this skip may be removed.

File: src/bindings/python/tests/test_runtime/test_core.py (hunk 16 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_core.py]: added

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 22]
```diff
@@ -335,18 +335,16 @@ def test_properties_core(properties_to_set):
 
     core.set_property(properties_to_set)
 
-    # RW properties without device name
-    assert core.get_property(properties.cache_dir()) == "./"
-    assert core.get_property(properties.force_tbb_terminate()) is False
+    device_name = core.get_property("CPU", "FULL_DEVICE_NAME")
+
+    if "Intel" not in device_name:
+        pytest.skip("Properties are Intel(R) CPU specific")
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py]: removed this check.

[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py]: about reflect in test name I think it's good idea, will change test names.

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 23]
```diff
@@ -274,7 +274,7 @@ def test_properties_hint_model():
 
 
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason=f"Cannot run test on device {os.environ.get('TEST_DEVICE')}, Plugin specific test")
-def test_single_property_setting():
+def test_single_cpu_property_setting():
     core = Core()
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_properties.py:278]: I suppose this test can be generic. Moreover, number of streams is supported by most (if not all) plugins. 
So, probably, we just need to remove limitations from original version of this test

File: src/bindings/python/tests/conftest.py [hunk 24]
```diff
@@ -89,7 +90,11 @@ def plugins_path():
     base_path = os.path.dirname(__file__)
     plugins_xml = os.path.join(base_path, "test_utils", "utils", "plugins.xml")
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/conftest.py:91]: can we parametrize this function `plugins_path` with device name, so the function can dynamically generate `plugins.xml` files with proper plugin file name?
In this case, we can even run the tests on GPU, GNA and other devices.

File: src/bindings/python/tests/test_runtime/test_core.py [hunk 25]
```diff
@@ -276,22 +255,25 @@ def test_query_model(device):
...
+    compiled_model = core.compile_model(model, "BLA")
+    assert isinstance(compiled_model, CompiledModel), "Cannot load the network to the registered plugin with name 'BLA'"
 
 
 @pytest.mark.dynamic_library()
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason="Device independent test")
-def test_register_plugins():
+def test_register_cpu_plugins():
     core = Core()
     if platform == "linux" or platform == "linux2":
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_core.py:278]: why `plugins_path` is not used here?
I suppose `plugins_path` needs to have not only arm / x86, but operating system as well.

Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_core.py]: have rewritten `plugins_path` function. Now it generates .xml file

File: src/bindings/python/tests/test_runtime/test_core.py [hunk 26]
```diff
@@ -276,22 +255,25 @@ def test_query_model(device):
...
 @pytest.mark.dynamic_library()
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason="Device independent test")
-def test_register_plugin():
+def test_register_cpu_plugin():
     core = Core()
-    core.register_plugin("openvino_intel_cpu_plugin", "BLA")
+    if "Intel" in core.get_property("CPU", "FULL_DEVICE_NAME"):
+        core.register_plugin("openvino_intel_cpu_plugin", "BLA")
+    else:
+        core.register_plugin("openvino_arm_cpu_plugin", "BLA")
```
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_core.py]: have rewritten `plugins_path`

File: src/bindings/python/tests/conftest.py (hunk 24 above)
Comment (akuporos) [reply src/bindings/python/tests/conftest.py]: have rewritten `plugins_path`

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 23 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_properties.py:277]: updated

Comment (ilya-lavrenov) [reply src/bindings/python/tests/test_runtime/test_properties.py:277]: but we still have skip here..

**Review by ilya-lavrenov — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]
LGTM

**Pull request merged**